### PR TITLE
[rac_kalasha_latn] New keyboard for Kalasha Latin

### PR DESCRIPTION
**Pull request by rachitralli**
[rac_kalasha_latn] please add Kalasha Keyboard in latin script

### CODE REVIEW

**Comment by rachitralli** [comment]
@DavidLRowe thank you very much for guidance, I have created new PR for this keyboard as desired, please see if you have time. regards

**Comment by keyman-server** [comment]
This pull request is from an external repo and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Comment by DavidLRowe** [comment]
Technical issues: 
Desktop keyboard: 
- .keyboard_info
  - file has too many fields; keep "license", "languages", and "description", then delete the rest (since they will be picked up when the keyboard is built), including "name", which seems to be incorrect (corresponds to incorrect language tag)
  - "languages" has a language tag (kxp-Arab) that seems incorrect: perhaps kls-Latn is what you want?
  - example is in Arabic script but this is a Latin script keyboard
- welcome.htm has instructions in Urdu in Arabic script. That seems odd for a Latin script keyboard.
- .kmn file
  - turn off RTL
  - reference "rac_kalasha_latn.keyman-touch-layout" instead of "rac_kalasha_latn-layout_latin.js"
- rac_kalasha_latn-layout_latin.js
  - rename to "rac_kalasha_latn.keyman-touch-layout" 
  - a number of corrections are needed

Touch layout:
- On both default layer and shift layer, the following longpress keys on the "." key have issues: ` ! ' " \ : `
- On the numeric layer the keys with the following labels don't produce what the key indicates: ` õʹ jʹ tʹ ~u ãʹ iʹ `

There are some concerns with the design of this keyboard: 
- How will a user type capital letters with tilde and/or `ʹ` ?
- Do you really need a separate key for each of the letters that combine with `ʹ` ? Couldn't you have a separate key for  `ʹ` and let users type a letter and then  `ʹ` ?
- Similarly, do `ts`, `zh` and `sh` need their own keys? or could users just type the two characters they need?
- You might even consider making the tilde a separate key that combines with a previously typed a, e, i, o or u. Otherwise, can you find space on the default layer for ã, ẽ, ĩ, õ, ũ and space on the shift layer for the capital versions?
- Can you provide a way for users to type the characters on keys that you replace with other letters?

**Comment by rachitralli** [comment]
@DavidLRowe the corrections in the Desktop keyboard has been done as desired by you. as far as the Touch layout please check the following:-

**Touch layout:**
I am unable to fix the longpress issues on the "." key have issues: ! ' " \ :
On the numeric layer the keys with the following labels don't produce what the key indicates: õʹ jʹ tʹ ~u ãʹ iʹ Is it expected? I am unable to fix this issue.
**How will a user type capital letters with tilde and/or ʹ ?**
when we type in word and press spacebar then the small letter will automatically be converted into capital letter.
**Do you really need a separate key for each of the letters that combine with ʹ ? Couldn't you have a separate key for ʹ and let users type a letter and then ʹ ?**
Yes I do really need a separate key for each of the letters that combine with ʹ
**Similarly, do ts, zh and sh need their own keys? or could users just type the two characters they need?**
the user just type the two characters they need, due to insufficient space there is no need separate key for ts, zh and sh, the users just type two characters they need.
**You might even consider making the tilde a separate key that combines with a previously typed a, e, i, o or u. Otherwise, can you find space on the default layer for ã, ẽ, ĩ, õ, ũ and space on the shift layer for the capital versions?**
no need capital version.

**Comment by DavidLRowe** [comment]
@rachitralli To fix issues with keys (or longpress keys) that don't produce what you want, you will likely want to change the key identifier ("Code" in Keyman Developer's Design view, or "id" in Code view). For example, the second longpress key on the "." key has a "Code" value of "K_1" with "Modifier" value of "shift". This links to what the .kmn file does with SHIFT K_1, that is, the "!". But in the .kmn file, you have mapped SHIFT K_1 to  "cʹ", so when you select the longpress key that shows "!" you actually get  "cʹ". 

The solution, in cases where you don't want to link to what is in the .kmn file, is to change the key identifier. There are two possibilities. If you only have one Unicode code point to output, you can use an identifier of the form U_abcd to output Unicode codepoint U+abcd. If you have multiple Unicode code points to output, you need to use an identifier of the form T_xxxxxx, and then match that with an entry in the .kmn file of the form + T_xxxxxx > U+abcd U+efgh . See the last section of this help document page:
https://help.keyman.com/developer/language/guide/virtual-keys

**Comment by rachitralli** [comment]
@DavidLRowe thank for suggestions, tried many time, but it is very complicated, let me try again. thanks

**Comment by rachitralli** [comment]
@DavidLRowe sorry sir, I am unable to fix the issue of longpress keys, I am requesting @MakaraSok to look into this issue as and when has some spare time. thanks

**Comment by DavidLRowe** [comment]
@rachitralli Did you get the email I sent with instructions on how you can address the problems with this pull request? Was anything not clear?

**Comment by rachitralli** [comment]
Sir due to covid 19 I am out of city till the end of September. Sorry sir unable to edit the same.

 With Profound Regards 

 

Rehmat Aziz Chitrali M.A., BLIS, LL.B, Law-Gat, DIL, DIJ
Gold Medalist(NPC),Gold Medalist(JFP),

WorldRecord Holder(Linguistics), 

InternationalInnovation Laureate(Germany), 

InnovativeInitiative Laureate, 

Linguist& Culture Researcher 

ShandoorAwardee, 

ChitralHuman Development Awardee

SAFETalent Award, 

JIYouth Pakistan Award 

PakistanYouth Icon Award(UN & YRC)

Prideof the Nation Award (JFP)

Dr.A.Q.KhanAward (NPC)

N-PeaceAward(Nominee)

JoharAward (Nominee)

AgahiAward(Nominee)

RisingStar Award

CommendationCertificate from Chief of Army Staff Gen Raheel Shareef

LinguisticAward  fromMyLanguages.Org

Tamgha-e-Pakistanfor National Merit

Sanad-e-Fazeelatfrom Dawah Academy, Isb. 

Sanad-e-Tauseeffrom NBF & Ministry of Education Govt of Pakistan 

Gold Medals.

1.Prideof the Nation Gold Medal(JFP)

2.Dr.AbdulQadeer Khan Gold Medal(NPC)

Honorary Degree.

1.HonoraryDegree of Associate Fellowship from Royal Commonwealth Society(RCS) GreatBritain.

World Record:

1.WorldRecord Holder in Linguistics

دیگر اعزازت، القابات و خطابات

۱۔پاکستانی زبانوں کا رکھوالا(روزنامہ جسارت)

۲۔۔ہندوکش کے پہاڑوں کے دامن میں چھپا ہیرا، رحمت عزیز چترالی(ماہنامہ روابطانٹرنیشنل)

3.RehmatAziz Chitrali(A man with a Vision) by Daily Tribune

Email:rachitrali@yahoo.com 

Books: http://www.scribd.com/khowaracademy

Keyboard:http://www.branah.com/khowar 
 

    On Thursday, August 13, 2020, 06:37:51 AM GMT+5, DavidLRowe <notifications@github.com> wrote:  
 
 


@rachitralli Did you get the email I sent with instructions on how you can address the problems with this pull request? Was anything not clear?

**Comment by DavidLRowe** [comment]
Thanks for the update. I wasn't trying to pressure you. I just wanted to make sure I wasn't preventing you from moving forward.

**Comment by LornaSIL** [comment]
@rachitralli since you haven't been able to update this PR I'm going to close the issue. You can submit a new PR when you are next able to do so.
